yaml/_dumper.ts

```diff
@@ -195,25 +195,20 @@ export class DumperState {
 
 function encodeHex(character: number): string {
   const string = character.toString(16).toUpperCase();
+  const length = string.length;
 
-  let handle: string;
-  let length: number;
   if (character <= 0xff) {
-    handle = "x";
-    length = 2;
-  } else if (character <= 0xffff) {
-    handle = "u";
-    length = 4;
-  } else if (character <= 0xffffffff) {
-    handle = "U";
-    length = 8;
-  } else {
-    throw new YamlError(
-      "code point within a string may not be greater than 0xFFFFFFFF",
-    );
+    return `\\x${"0".repeat(2 - length)}${string}`;
```

Comment:
Looks like this can be simplified further with `.padStart` method?

Reply:
You are right, I overlooked that one.